### PR TITLE
Fix Application Output hang when using ASP.Net Core Angular template

### PR DESCRIPTION
**Pull request by iantoalms**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/893213

Added method in the LogView to sanitize the logging text output from the node.js service for the Angular/Bootstrap functionality in the ASP.Net Core Angular template.

The logging text contained ANSI/VT100 color escape sequences and some control characters, which were causing the pango text engine to slow down significantly in order to create custom glyphs with the control codes.

### CODE REVIEW

**Comment by rodrmoya** [comment]
@monojenkins backport release-8.1

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/LogView.cs [hunk 1]
```diff
@@ -693,6 +696,23 @@ bool ShouldAutoScroll ()
...
+		///
+		/// Specifically we remove color escape sequences (\e[ + color code + m) and the backspace (\b) control
+		/// character.
+		/// 
+		/// </summary>
+		/// <param name="text"></param>
+		/// <returns>original text with control characters and escape sequences removed</returns>
+		string SanitizeConsoleText(string text)
+		{
+			return Regex.Replace (text, "(\\e[[0-9]*m)|(\b)", string.Empty);
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/LogView.cs]: Can we compile this regex into a static instance?

Comment (Therzok) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/LogView.cs]: I assume this is going to be used a lot, so having a compiled regex would result in faster processing.

Comment (iantoalms) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/LogView.cs]: Good idea. Done.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/LogView.cs [hunk 2]
```diff
@@ -58,6 +59,7 @@ public class LogView : Gtk.VBox
 		QueuedTextWrite lastTextWrite;
 		GLib.TimeoutHandler outputDispatcher;
 		bool outputDispatcherRunning = false;
+		readonly Regex consoleTextSanitizerRegex = new Regex ("(\\e[[0-9]*m)|(\b)");
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/LogView.cs]:
```suggestion
		readonly Regex consoleTextSanitizerRegex = new Regex ("(\\e[[0-9]*m)|(\b)", RegexOptions.Compiled);
```

**Comment by rodrmoya** [comment]
@monojenkins backport release-8.1

**Pull request merged**